app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,23 @@
+class CommentsController < ApplicationController
+  def new
+    @post = @current_user.posts.find(params[:post_id])
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.new(comment_params)
+    @comment.author = @user
+    if @comment.save
+      redirect_to user_post_path(@user, @post)
+    else
+      render :new
+    end
+  end
```

Comment:
- [ ] Please make sure to use the created `current_user` method in the application_controller file as per the project requirements. This will ensure that the user is accessible easily by the controllers.
`@comment = @current_user.comments.new(comment_params`
`@comment.post_id = params[:post_id]`